### PR TITLE
Civil Tongue: Add filtering to activity & notifications

### PR DESCRIPTION
**Pull request by slafleche**
Hooked into "beforeSave" to filter out activity messages in Civil Tongue plugin.

Closes https://github.com/vanilla/addons/issues/467

### CODE REVIEW

File: plugins/CivilTongueEx/class.civiltongueex.plugin.php [hunk 1]
```diff
@@ -379,6 +379,18 @@ public function notificationsController_informNotifications_handler($Sender, &$A
     }
 
     /**
+     * Cleanup Activity messages.
+     *
+     * @param $Sender
+     * @param $Args
+     */
+    public function ActivityModel_BeforeSave_Handler($Sender, &$Args) {
```
Comment (linc) [inline plugins/CivilTongueEx/class.civiltongueex.plugin.php]: camelCase not CapitalCase.

**Comment by slafleche** [comment]
Yes I did test 
screen shot 2017-06-02 at 08 17 51

**Comment by linc** [comment]
Look at the method names above and below this one and try again.

**Comment by linc** [comment]
So, this is working because the value at the key of 'Activity' in this case happens to be an object, which is passed by reference. In all other cases (including if 'Activity' happened to be formatted as an array) this would fail because you're just setting in a value that is never returned or set on $Args. And, despite using `val()` as an abstraction (which is so we don't care if this is object or array) this makes it care - it explicitly breaks anyway as an array.

The correct way to close this loop is to create a third line that sets the value of `Activity` back on the args. But I would do this the more common way, which is to set `$sender->EventArguments['Activity'] = $activity;` manually rather than passing the `$Args` param by reference. We don't pass params as a reference anymore and it's bad to rely on that.

**Pull request merged**